test/compilable/__mutable.d

```diff
@@ -0,0 +1,92 @@
+// REQUIRED_ARGS: -o-
```

Comment:
Why disable the object generation?
I think it wouldn't hurt to have at least one test that does test that `__mutable` interacts fine without the backend failing with it.